File: samples/sample.java
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.stream.Collectors;
import java.util.stream.Stream;

// Record (Java 14+)
record Person(String name, int age, String email) {}

// Sealed interface (Java 17+)
sealed interface Shape permits Circle, Rectangle {
    double area();
}

// Pattern matching (Java 21+)
record Circle(double radius) implements Shape {
    @Override public double area() { return Math.PI * radius * radius; }
}

record Rectangle(double width, double height) implements Shape {
    @Override public double area() { return width * height; }
}

public class CompleteJavaFeatures {
    // Enums
    enum Color { RED, GREEN, BLUE }
    
    // Nested class
    static class Counter {
        private int count = 0;
        public synchronized void increment() { count++; }
        public int getCount() { return count; }
    }
    
    // Main method with var (Java 10+)
    public static void main(String[] args) {
        System.out.println("=== Java Complete Features Demo ===\n");
        
        // 1. PRIMITIVES & WRAPPERS
        int primitiveInt = 42;
        Integer wrapperInt = 42; // Autoboxing
        System.out.println("Primitives: " + primitiveInt + ", Wrapper: " + wrapperInt);
        
        // 2. ARRAYS
        int[] array = {1, 2, 3};
        System.out.println("Array: " + Arrays.toString(array));
        
        // 3. COLLECTIONS
        List<String> list = new ArrayList<>(List.of("apple", "banana", "cherry"));
        Set<Integer> set = new HashSet<>(Set.of(1, 2, 2, 3)); // Factory methods (Java 9+)
        Map<String, Integer> map = new HashMap<>(Map.of("a", 1, "b", 2));
        System.out.println("List: " + list + ", Set: " + set + ", Map: " + map);
        
        // 4. STRINGS & TEXT BLOCKS (Java 15+)
        String textBlock = """
            This is a multi-line
            text block with %s!
            """.formatted("Java");
        System.out.println("Text block:\n" + textBlock);
        
        // 5. RECORDS
        var person = new Person("Alice", 30, "alice@example.com");
        System.out.println("Record: " + person);
        
        // 6. LAMBDA & FUNCTIONAL INTERFACES (Java 8+)
        List<Integer> numbers = List.of(1, 2, 3, 4, 5);
        List<Integer> doubled = numbers.stream()
            .map(n -> n * 2)
            .collect(Collectors.toList());
        System.out.println("Stream doubled: " + doubled);
        
        // Method references
        list.sort(String::compareTo);
        System.out.println("Sorted: " + list);
        
        // 7. OPTIONAL
        Optional<String> optional = Optional.ofNullable(getNameOrNull());
        String name = optional.orElse("Unknown");
        System.out.println("Optional: " + name);
        
        // 8. SWITCH EXPRESSIONS (Java 14+)
        Color color = Color.RED;
        String colorName = switch (color) {
            case RED -> "Red";
            case GREEN -> "Green";
            case BLUE -> "Blue";
        };
        System.out.println("Switch expr: " + colorName);
        
        // 9. PATTERN MATCHING (Java 21+)
        Object obj = new Circle(5);
        if (obj instanceof Circle c) {
            System.out.println("Circle area: " + c.area());
        }
        
        // 10. GENERICS
        class Box<T> {
            private T value;
            Box(T value) { this.value = value; }
            T get() { return value; }
        }
        Box<String> stringBox = new Box<>("Hello");
        System.out.println("Generic: " + stringBox.get());
        
        // 11. EXCEPTIONS
        try {
            riskyOperation();
        } catch (Exception e) {
            System.out.println("Caught: " + e.getMessage());
        }
        
        // 12. THREADS & VIRTUAL THREADS (Java 21+)
        try {
            ExecutorService executor = Executors.newVirtualThreadPerTaskExecutor();
            executor.submit(() -> System.out.println("Virtual thread: " + 
                Thread.currentThread()));
            executor.shutdown();
        } catch (Exception e) {
            e.printStackTrace();
        }
        
        // 13. CONCURRENCY
        Counter counter = new Counter();
        Runnable task = () -> {
            for (int i = 0; i < 1000; i++) counter.increment();
        };
        Thread[] threads = new Thread[10];
        for (int i = 0; i < 10; i++) {
            threads[i] = new Thread(task);
            threads[i].start();
        }
        for (Thread t : threads) t.join();
        System.out.println("Counter: " + counter.getCount());
        
        // 14. STREAM OPERATIONS
        long count = numbers.stream()
            .filter(n -> n % 2 == 0)
            .mapToLong(n -> n)
            .sum();
        System.out.println("Even sum: " + count);
        
        // 15. LOCAL TYPE INFERENCE (var)
        var list2 = new ArrayList<Integer>();
        var map2 = Map.of("key", "value");
        var stream = Stream.of(1, 2, 3);
        
        // 16. ENUM WITH METHODS
        System.out.println("Enum ordinal: " + Color.RED.ordinal());
        
        // 17. DATETIME API (Java 8+)
        LocalDateTime now = LocalDateTime.now();
        System.out.println("Now: " + now.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME));
    }
    
    static String getNameOrNull() { return null; }
    
    static void riskyOperation() throws Exception {
        throw new Exception("Test exception");
    }
    
    // Inner class
    class Inner {
        void sayHello() {
            System.out.println("Hello from inner: " + CompleteJavaFeatures.this);
        }
    }
    
    // Static nested class access
    static void demoInner() {
        CompleteJavaFeatures outer = new CompleteJavaFeatures();
        Inner inner = outer.new Inner();
        inner.sayHello();
    }
    
    // Functional interface example
    @FunctionalInterface
    interface Calculator {
        int calculate(int a, int b);
    }
    
    static void demoFunctional() {
        Calculator add = (a, b) -> a + b;
        System.out.println("Functional: " + add.calculate(2, 3));
    }
}

File: samples/sample.py
#!/usr/bin/env python3
"""
Complete Python Features Demo - Python 3.12+
Shows every major language feature in one file
"""

import asyncio
import dataclasses
import enum
import functools
import itertools
import json
import multiprocessing as mp
import os
import re
import sys
import threading
import time
import typing
from collections import defaultdict, deque, namedtuple
from concurrent.futures import ThreadPoolExecutor
from datetime import datetime, timedelta
from pathlib import Path
from typing import (
    Any, Callable, Dict, Generic, List, Literal, Optional, Protocol, TypeAlias, TypeVar
)
from weakref import WeakKeyDictionary

# ===== 1. BASIC TYPES & LITERALS =====
PRIMITIVE_INT: int = 42
PRIMITIVE_FLOAT: float = 3.14159
BOOL_TRUE: bool = True
BOOL_FALSE: bool = False
NONE_VAL: None = None
IMAGINARY: complex = 1 + 2j

# Type aliases (Python 3.10+)
Point: TypeAlias = tuple[float, float]
UserId: TypeAlias = int

# ===== 2. COLLECTIONS =====
LIST_VAL: List[int] = [1, 2, 3]
TUPLE_VAL: tuple[int, str, float] = (1, "hello", 3.14)
DICT_VAL: Dict[str, int] = {"a": 1, "b": 2}
SET_VAL: set[str] = {"apple", "banana"}

# NamedTuple
PersonNT = namedtuple("PersonNT", ["name", "age"])
person_nt = PersonNT("Alice", 30)

# defaultdict
counter = defaultdict(int)

# deque
queue = deque([1, 2, 3])

# ===== 3. STRINGS & FORMATTING =====
SIMPLE_STR = "hello"
FSTRING = f"Value: {PRIMITIVE_INT}"
RAW_STR = r"C:\path\to\file"
BYTES_STR = b"bytes"
TRIPLE_QUOTE = """
Multi-line
string
"""

# ===== 4. CLASSES & OBJECTS =====
class BaseClass:
    def __init__(self, value: int):
        self._value = value
    
    @property
    def value(self) -> int:
        return self._value
    
    @classmethod
    def create(cls) -> 'BaseClass':
        return cls(42)

class DerivedClass(BaseClass):
    def method(self) -> str:
        return f"Derived: {self.value}"

# Dataclasses (Python 3.7+)
@dataclasses.dataclass(frozen=True, order=True)
class Person:
    name: str
    age: int
    email: str = dataclasses.field(default="no-email@example.com")

# Protocol (structural typing)
class Drawable(Protocol):
    def draw(self) -> None: ...

class Circle:
    def draw(self) -> None:
        print("Drawing circle")

# Generic classes
T = TypeVar('T')
class Stack(Generic[T]):
    def __init__(self):
        self._items: List[T] = []
    
    def push(self, item: T) -> None:
        self._items.append(item)
    
    def pop(self) -> T:
        return self._items.pop()

# ===== 5. ENUMS =====
class Color(enum.Enum):
    RED = "red"
    GREEN = "green"
    BLUE = "blue"

class Status(enum.Flag):
    NONE = 0
    ACTIVE = 1
    DELETED = 2

# ===== 6. FUNCTIONS & DECORATORS =====
def simple_func(param1: int, param2: str = "default") -> tuple[int, str]:
    return (param1 * 2, param2)

@functools.lru_cache(maxsize=128)
def fibonacci(n: int) -> int:
    if n <= 1:
        return n
    return fibonacci(n-1) + fibonacci(n-2)

def generator_func():
    yield 1
    yield 2
    yield 3

async def async_func() -> str:
    await asyncio.sleep(0.1)
    return "async complete"

# Lambda
square = lambda x: x * x

# Higher-order function
def apply_twice(func: Callable[[int], int], x: int) -> int:
    return func(func(x))

# ===== 7. EXCEPTIONS =====
class CustomError(Exception):
    """Custom exception class"""

# ===== 8. CONTEXT MANAGERS =====
class FileManager:
    def __enter__(self):
        self.file = open("temp.txt", "w")
        return self.file
    
    def __exit__(self, exc_type, exc_val, exc_tb):
        self.file.close()

# ===== 9. MAIN EXECUTION =====
def main():
    print("=== PYTHON COMPLETE FEATURES DEMO ===\n")
    
    # 1. Basic types
    print(f"Primitives: {PRIMITIVE_INT}, {PRIMITIVE_FLOAT}, {BOOL_TRUE}, {NONE_VAL}")
    print(f"Imaginary: {IMAGINARY}")
    
    # 2. Collections
    print(f"List: {LIST_VAL}, Tuple: {TUPLE_VAL}")
    print(f"Dict: {DICT_VAL}, Set: {SET_VAL}")
    print(f"NamedTuple: {person_nt}")
    
    # 3. Strings
    print(f"F-string: {FSTRING}")
    print(f"Textwrap: {TRIPLE_QUOTE}")
    
    # 4. Classes & dataclasses
    person = Person("Bob", 25)
    print(f"Dataclass: {person}, Age: {person.age}")
    
    base = BaseClass(100)
    derived = DerivedClass(200)
    print(f"Class inheritance: {derived.method()}")
    
    # 5. Enums
    print(f"Enum: {Color.RED.value}")
    
    # 6. Functions
    print(f"Simple func: {simple_func(5)}")
    print(f"Fibonacci(10): {fibonacci(10)}")
    
    gen = generator_func()
    print(f"Generator: {list(gen)}")
    
    # 7. Type hints & generics
    int_stack = Stack[int]()
    int_stack.push(42)
    print(f"Generic stack: {int_stack.pop()}")
    
    # 8. List comprehensions & generators
    squares = [x**2 for x in range(5)]
    gen_expr = sum(x**2 for x in range(5))
    print(f"List comp: {squares}, Gen expr: {gen_expr}")
    
    # 9. Pattern matching (Python 3.10+)
    def match_value(value):
        match value:
            case int(x) if x > 0:
                return "positive int"
            case str() as s if len(s) > 3:
                return "long string"
            case _:
                return "other"
    
    print(f"Pattern match 42: {match_value(42)}")
    print(f"Pattern match 'hello': {match_value('hello')}")
    
    # 10. Decorators
    @functools.singledispatch
    def process(value):
        return f"Unknown: {value}"
    
    @process.register
    def _(value: int):
        return f"Int: {value}"
    
    print(f"Single dispatch: {process(123)}")
    
    # 11. Concurrency
    print("Threading:")
    def thread_worker(n):
        time.sleep(0.1)
        return n * 2
    
    with ThreadPoolExecutor(max_workers=2) as executor:
        futures = [executor.submit(thread_worker, i) for i in range(3)]
        results = [f.result() for f in futures]
    print(f"Thread results: {results}")
    
    # 12. Multiprocessing
    print("Multiprocessing:")
    def mp_worker(x):
        return x * x
    
    with mp.Pool(2) as pool:
        mp_results = pool.map(mp_worker, [1, 2, 3])
    print(f"MP results: {mp_results}")
    
    # 13. Async/Await
    print("Async (non-blocking):")
    async def fetch_data():
        await asyncio.sleep(0.1)
        return "data"
    
    async def main_async():
        tasks = [fetch_data() for _ in range(3)]
        results = await asyncio.gather(*tasks)
        print(f"Async results: {results[:1]}...")
    
    asyncio.run(main_async())
    
    # 14. pathlib & modern file handling
    path = Path("example.txt")
    path.write_text("Hello, Python!")
    content = path.read_text()
    print(f"Pathlib: {content[:10]}...")
    
    # 15. Walrus operator (Python 3.8+)
    data = [1, 2, 3, 4, 5]
    if (n := len(data)) > 3:
        print(f"Walrus: list has {n} items")
    
    # 16. Union types & Literal (Python 3.10+)
    def greet(mode: Literal["formal", "casual"]) -> str:
        return {"formal": "Good day", "casual": "Hey"}[mode]
    
    print(f"Literal type: {greet('formal')}")
    
    print("\n=== All Python features demo complete! ===")

if __name__ == "__main__":
    # Global setup
    counter["demo"] += 1
    main()